### PR TITLE
Add pagination function and fix timer bug

### PR DESCRIPTION
**Pull request by huanghao1998**
1. Fix the timer bug #60 
2. In this update, during matching can leave the page, the connection will not lost #59
3. Add pagination for table and fixed the table size 
4. Alert component now at the top layer, so even changing page can still alert
5. Add update button for upcoming serverless function
6. Navbar now always stick to the top
7. Able to change language in the code editor page
image

### CODE REVIEW

File: apps/frontend/src/components/Countdown.tsx [hunk 1]
```diff
@@ -10,8 +10,10 @@ type CountdownProps = {
 
 const Countdown: React.FC<CountdownProps> = (
     {counter}
-    ) => {
-
+) => {
+  useEffect(() => {
+    console.log(counter);
+  }, [counter]);
```
Comment (igezt) [inline apps/frontend/src/components/Countdown.tsx]: Is this line still needed?

File: apps/frontend/src/components/questions/questionPage/codeEditor/EditorNav.tsx [hunk 2]
```diff
@@ -1,19 +1,45 @@
...
+                <li
+                  key={language}
+                  className="cursor-pointer hover:bg-gray-200 p-2 text-xs rounded-md"
+                  onClick={() => selectLanguage(language)}
+                >
+                  {language}
+                </li>
+              ))}
+            </ul>
+          )}
```
Comment (igezt) [inline apps/frontend/src/components/questions/questionPage/codeEditor/EditorNav.tsx:42]: Lit 🔥

File: apps/frontend/src/components/questions/questionTable/QuestionPagination.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+              <span className="sr-only">Next</span>
+              <ChevronRightIcon className="h-5 w-5" aria-hidden="true" />
+            </button>
+          </nav>
+        </div>
+      </div>
+    </div>
+  );
+};
+export default QuestionPagination;
```
Comment (igezt) [inline apps/frontend/src/components/questions/questionTable/QuestionPagination.tsx:122]: Good stuff with the pagination!

**Review by igezt — APPROVED** [review]
Looks good! Just need to make a small update with one of the log statement

**Review by Jarrett0203 — COMMENTED** [review]
not sure if theres something wrong on my side, but I get this error when signing in.

unhandledRejection: ReferenceError: window is not defined
[4]     at _callee6$ (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\next-auth\react\index.js:254:160)
[4]     at tryCatch (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\regeneratorRuntime.js:45:16)
[4]     at Generator.<anonymous> (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\regeneratorRuntime.js:133:17)
[4]     at Generator.next (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\regeneratorRuntime.js:74:21)
[4]     at asyncGeneratorStep (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\asyncToGenerator.js:3:24)
[4]     at _next (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\asyncToGenerator.js:22:9)
[4]     at C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\asyncToGenerator.js:27:7
[4]     at new Promise (<anonymous>)
[4]     at C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\asyncToGenerator.js:19:12
[4]     at signIn (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\next-auth\react\index.js:243:18)
[4]     at Layout (webpack-internal:///./src/components/Layout.tsx:39:64)
[4]     at renderWithHooks (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5658:16)       
[4]     at renderIndeterminateComponent (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5731:15)
[4]     at renderElement (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5946:7)
[4]     at renderNodeDestructiveImpl (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\react-dom\cjs\react-dom-server.browser.development.js:6104:

**Comment by huanghao1998** [comment]
> not sure if theres something wrong on my side, but I get this error when signing in.
> 
> 
> 
> unhandledRejection: ReferenceError: window is not defined
> 
> [4]     at _callee6$ (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\next-auth\react\index.js:254:160)
> 
> [4]     at tryCatch (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\regeneratorRuntime.js:45:16)
> 
> [4]     at Generator.<anonymous> (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\regeneratorRuntime.js:133:17)
> 
> [4]     at Generator.next (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\regeneratorRuntime.js:74:21)
> 
> [4]     at asyncGeneratorStep (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\asyncToGenerator.js:3:24)
> 
> [4]     at _next (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\asyncToGenerator.js:22:9)
> 
> [4]     at C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\asyncToGenerator.js:27:7
> 
> [4]     at new Promise (<anonymous>)
> 
> [4]     at C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\@babel\runtime\helpers\asyncToGenerator.js:19:12
> 
> [4]     at signIn (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\next-auth\react\index.js:243:18)
> 
> [4]     at Layout (webpack-internal:///./src/components/Layout.tsx:39:64)
> 
> [4]     at renderWithHooks (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5658:16)       
> 
> [4]     at renderIndeterminateComponent (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5731:15)
> 
> [4]     at renderElement (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\react-dom\cjs\react-dom-server.browser.development.js:5946:7)
> 
> [4]     at renderNodeDestructiveImpl (C:\Users\Jarrett\Desktop\nus\Y3S1\CS3219\ay2324s1-course-assessment-g48\node_modules\react-dom\cjs\react-dom-server.browser.development.js:6104:

I think you can try rebuild your nextjs using yarn build as i also encounter this before

File: apps/frontend/src/components/Countdown.tsx (hunk 1 above)
Comment (huanghao1998) [reply apps/frontend/src/components/Countdown.tsx]: Oh ****, i forget to remove haha

**Pull request merged**